### ISSUE TITLE
eSIM Cannot be Disabled from PIN Screen

### ISSUE THREAD

**Issue by inferenceus** [GrapheneOS/os-issue-tracker#2162]
After a reboot to update to GrapheneOS version 2023042900 (unsure if version-related or generic to the OS), I have forgotten my eSIM PIN and attempting to disable it via the "Disable eSIM" button throws the following error:
```
Can't disable eSIM
The eSIM can't be disabled due to an error.
```

Is this a bug, or a known issue/feature of GrapheneOS? Possibly due to the privileged eSIM management toggle being disabled on reboot?

**Comment by inferenceus** [comment]
I was lucky enough to find my PUK code for the eSIM, which allowed me to unlock the eSIM after the 3 failed attempts of standard PIN unlocking.

If this is intentional behaviour, possibly due to the privileged eSIM management, it should be stated on the GrapheneOS website (I can't find any such notice of this behaviour there), and the `Disable eSIM` button possibly removed from the SIM PIN screen.

If not for finding the PUK code, I would have had to purchase a new device just to be able to contact my carrier, since a physical SIM would likely have asked for the eSIM PIN since it would still installed alongside the physical SIM.

**Comment by maade69** [comment]
Yes, privileged eSIM management should be enabled if you want to disable eSIM. I believe that 

> However, in order to manage and add eSIMs, proprietary Google functionality is needed.
 in [Usage section](https://grapheneos.org/usage#sandboxed-google-play-esim) covers that, but the removal of eSIM should be more detailed

**Comment by surfaceflinger** [comment]
There should be some way of dismissing this prompt.
I had exactly the same situation 2 days ago. I don't even want to check what would happen if I entered wrong PUK 10 times for science.

In my case I triggered it by entering the wrong PIN or purpose, I use a carrier that's 100% managed from app, for eSIM they don't provide PIN, only initial PUK. I wanted to enable PIN so I entered wrong PIN 3 times to then set my own PIN using PUK.
The issue was that I didn't write down the PUK from app, I just assumed that I could simply go to Recents and write it down from there.

In this state the device is practically unusable.

(oh, and yes, I fixed it by downloading carrier app on another phone)

**Comment by domnru** [comment]
I'm having the same problem.

@surfaceflinger 
> I don't even want to check what would happen if I entered wrong PUK 10 times for science.

I did, and it wasn't fun. I have a second phone with GOS on it. This resulted in my older Pixel 6a getting stuck in an error loop. After a factory reset, the PUK prompt was displayed again. I had to fix it on the phone by removing GOS, installing the stock operating system, disabling the eSIM, and then installing GOS again.

But on my current phone, I don't want to wipe all of my data. I have most of my important data backed up, but that doesn't matter in this situation.

To fix this, GOS should either enable "Google eSIM management" in this screen or allow the user to decide. Disabling the PUK screen until the next reboot would also be a good solution. Most carriers today (especially for eSIMs) allow users to view their PUK codes in an app. However, if you don't have your PUK code and no second device (or in my case, I have 2FA on the phone number that is locked, which is ridiculous), you will be completely locked out of your data. The only way to get rid of the eSIM is to completely wipe all data with the eSIM installed and install the stock ROM and disable it there.

Another option would be to enable the ability to disable the eSIM system completely via adb or sideload in Android rescue mode. Alternatively, in the fastboot menu, a feature to disable the SIM system altogether would be a valuable security measure.

**Comment by surfaceflinger** [comment]
awesome, thanks!
